cmd/admin-handlers.go

```diff
@@ -284,6 +284,73 @@ func (a adminAPIHandlers) ServerInfoHandler(w http.ResponseWriter, r *http.Reque
 	writeSuccessResponseJSON(w, jsonBytes)
 }
 
+// ServerDrivePerfInfo holds informantion about address and write speed of all drives in a single server node
+type ServerDrivePerfInfo struct {
+	Addr       string            `json:"addr"`
+	DrivesPerf []DriveWriteSpeed `json:"drivesPerf"`
+}
+
+// DriveWriteSpeed holds informantion about write speed of a single drive
+type DriveWriteSpeed struct {
+	Path       string  `json:"path"`
+	WriteSpeed float64 `json:"writeSpeed"`
+}
+
+// DriveWriteInfoHandler - GET /minio/admin/v1/health/drive
+// ----------
+// Get server information
+func (a adminAPIHandlers) DriveWritePerfHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "DriveWritePerfInfo")
+
+	// Authenticate request
+	// Setting the region as empty so as the mc server info command is irrespective to the region.
+	adminAPIErr := checkAdminRequestAuthType(ctx, r, "")
+	if adminAPIErr != ErrNone {
+		writeErrorResponseJSON(w, adminAPIErr, r.URL)
+		return
+	}
+
+	// Web service response
+	reply := make([]ServerDrivePerfInfo, len(globalAdminPeers))
+
+	var wg sync.WaitGroup
+
+	// Gather server information for all nodes
+	for i, p := range globalAdminPeers {
+		wg.Add(1)
+
+		// Gather information from a peer in a goroutine
+		go func(idx int, peer adminPeer) {
+			defer wg.Done()
+
+			// Initialize server info at index
+			reply[idx] = ServerDrivePerfInfo{Addr: peer.addr}
+
+			drivesPerf, err := peer.cmdRunner.ServerDrivePerfInfo()
```

Comment:
Use the notification.go for this we are going to deprecate the user of peer.cmdRunner @nitisht